### DISCUSSION TITLE
Syntax Highlighting broken for ERB when viewing a full file?

### DISCUSSION THREAD

**Discussion by thewatts** [community/community#62428] [Repositories]
### Select Topic Area

Bug

### Body

Hi there! 😄

I noticed while going through some of our pull requests - that syntax highlighting for `.erb` files and when viewing full files - syntax highlighting seems to be broken (?).

Here's an example file: 
https://gist.github.com/thewatts/166c379ddb7298a4195599b32359aa1b

Here's a screenshot of what it looks like full file:
<img width="1052" alt="CleanShot 2023-08-01 at 21 37 48@2x" src="https://github.com/community/community/assets/1141717/0ebe1a77-0622-4aea-99b2-612e7da8ce6c">

But you can see, with the `gist` preview, it's highlighting correctly:
<img width="948" alt="CleanShot 2023-08-01 at 21 38 11@2x" src="https://github.com/community/community/assets/1141717/383b093c-df69-414a-8d1f-e502468ab5dc">

---

Additional Context:

It looks like in certain instances, wrapping in a `div` fixes the problem:
https://gist.github.com/thewatts/ca236c59934820a80d5ac0b6b9b662bf

<img width="850" alt="CleanShot 2023-08-01 at 22 25 33@2x" src="https://github.com/community/community/assets/1141717/0634587f-fd2f-46c7-8dba-0efe1955c466">

But in other cases, it's not a problem at all:
https://gist.github.com/thewatts/86d51ce265eb383f1a332f70d6816245

<img width="1022" alt="CleanShot 2023-08-01 at 22 27 12@2x" src="https://github.com/community/community/assets/1141717/f8fd9778-4d53-4763-acc8-059725cbe5a2">

---

Thanks so much for your help!

**Comment by h7x4** [comment]
I don't know much about this language, so I can't really tell if the syntax highlighting is reacting that way based on a specific error. But if you think there's a bug in the syntax highlighter, I think you should raise an issue at https://github.com/github-linguist/linguist

**Reply by h7x4** [reply]
There seems to be an existing issue about erb there:

https://github.com/github-linguist/linguist/issues/5629

According to the comments, erb is not a html only format, but it can contain anything. I suspect, and this is just a guess, that the highlighting is trying to be smart by trying to determine the "inner" language by looking for some keywords. Like it would assume it's html if it starts with a `<div>`.

It also seems like if you name your files like `.html.erb` or `.js.erb`, it might be easier for it to determine the language.
